src/D2L.Security.OAuth2/Keys/Default/D2LSecurityToken.cs

```diff
@@ -27,96 +23,80 @@ internal sealed partial class D2LSecurityToken : SecurityToken {
 				throw new ArgumentException( "validFrom must be before validTo" );
 			}
 
-			m_id = id;
-			m_validFrom = validFrom;
-			m_validTo = validTo;
+			Id = id;
+			ValidFrom = validFrom;
+			ValidTo = validTo;
 
 			m_key = new ThreadLocal<Tuple<AsymmetricSecurityKey, IDisposable>>(
-				valueFactory: keyFactory,
+				valueFactory: () => {
+					var result = keyFactory();
+					result.Item1.KeyId = KeyId;
+					return result;
+				},
 				trackAllValues: true
 			);
 		}
 
-		public string KeyId { get { return m_id; } }
-
-		public override DateTime ValidFrom {
-			get { return m_validFrom.UtcDateTime; }
-		}
+		public string Id { get; }
+		public DateTimeOffset ValidFrom { get; }
+		public DateTimeOffset ValidTo { get; }

```

Comment:
- [ ] I don't believe anything checks this anymore. Need to assess checking dates before using the key.
  - [ ] Compare to old library

Reply:
Initial assessment says the old library didn't look at these either. So we'll sign/verify with keys that are out of range.

I believe our key cache stuff negates this, but, something to assess / throw an error for down the line?

Reply:
Ouch, yeah :(